functions/Backup-DbaDatabase.ps1

```diff
@@ -356,6 +387,33 @@ Backs up AdventureWorks2014 to sql2016's C:\temp folder
 					$Filelist = @()
 					$FileList += $server.Databases[$dbname].FileGroups.Files | Select-Object @{Name="FileType";Expression={"D"}}, @{Name="LogicalName";Expression={$_.Name}}, @{Name="PhysicalName";Expression={$_.FileName}}
 					$FileList += $server.Databases[$dbname].LofFiles | Select-Object @{Name="FileType";Expression={"L"}}, @{Name="LogicalName";Expression={$_.Name}}, @{Name="PhysicalName";Expression={$_.FileName}}
+					if ($Verify)
+					{
+						$verifiedresult = [PSCustomObject]@{
+										SqlInstance = $server.name
+										DatabaseName = $dbname
+										BackupComplete = $BackupComplete
+										BackupFilesCount = $FinalBackupPath.count	
+										BackupFile = (split-path $FinalBackupPath -leaf)
+										BackupFolder = (split-path $FinalBackupPath | Sort-Object -Unique)
+										BackupPath = ($FinalBackupPath | Sort-Object -Unique)
+										Script = $script
+										Notes = $failures -join (',')
+										FullName = ($FinalBackupPath | Sort-Object -Unique)
+										FileList = $FileList
+										SoftwareVersionMajor = $server.VersionMajor
+								}  | Restore-DbaDatabase -SqlServer $server.name -SqlCredential $SqlCredential -DatabaseName DbaVerifyOnly -VerifyOnly
```

Comment:
Restore-DbaDatabase accepts objects. This means that you can pass the full SMO object of just $server instead of initiating a new connection.